src/lightning/fabric/plugins/precision/amp.py

```diff
@@ -50,15 +51,19 @@ def __init__(
         self.device = device
         self.scaler = scaler
 
+        precision_to_type = {"bf16-mixed": torch.bfloat16, "16-mixed": torch.float16}
+        self._desired_dtype = precision_to_type[self.precision]
```

Comment:
We could make this public API

```suggestion
        self.dtype = precision_to_type[self.precision]
```

Reply:
I'd be fine with `self._dtype` to keep it internal.

Given your reasoning, then this should be `self._inputs_dtype`, "desired" could be for inputs/weights, ...